### ISSUE TITLE
Intermittent TypeError in load balancer handlers

### ISSUE THREAD

**Issue by SamCB-CT** [grpc/grpc-node#2379]
### Problem description

Since updating a few google-cloud sdks, resulting in gRPC updates, we've been receiving intermittent errors with the following stack trace.

```
/app/node_modules/@grpc/grpc-js/build/src/subchannel-address.js:22
    return 'port' in address;
                  ^

TypeError: Cannot use 'in' operator to search for 'port' in undefined
    at isTcpSubchannelAddress (/app/node_modules/@grpc/grpc-js/build/src/subchannel-address.js:22:19)
    at subchannelAddressEqual (/app/node_modules/@grpc/grpc-js/build/src/subchannel-address.js:26:9)
    at /app/node_modules/@grpc/grpc-js/build/src/load-balancer-pick-first.js:329:109
    at Array.every (<anonymous>)
    at PickFirstLoadBalancer.updateAddressList (/app/node_modules/@grpc/grpc-js/build/src/load-balancer-pick-first.js:329:37)
    at ChildLoadBalancerHandler.updateAddressList (/app/node_modules/@grpc/grpc-js/build/src/load-balancer-child-handler.js:108:23)
    at Object.onSuccessfulResolution (/app/node_modules/@grpc/grpc-js/build/src/resolving-load-balancer.js:177:40)
    at /app/node_modules/@grpc/grpc-js/build/src/resolver-dns.js:187:31
```

These errors have been seen across numerous containers, numerous environments in small bursts within a few minutes of each other.

May be related: https://github.com/grpc/grpc-node/pull/2369

We've since pinned backed to 1.8.7.

### Reproduction steps
When upgrading from 1.8.7 to 1.8.11, because of upgrades to:
- `@google-cloud/pubsub`
- `@google-cloud/speech`
- `@google-cloud/translate`

Type errors intermittently occur. 

### Environment
 - OS name, version and architecture: Kubernetes 
 - Node version 16.19.1
 - Node installation method - docker: node:16-alpine
 - Package name and version - @grpc/grpc-js@1.8.11

**Comment by murgatroid99** [comment]
I published a fix for this in version 1.8.12. Please try it out.

**Comment by lawrenceong** [comment]
Confirmed there are no issues for about 2 weeks since fix was made.  Thanks @murgatroid99.
